artifacts/scripting/functions.yml

```diff
@@ -0,0 +1,1483 @@
+  items:
+  - name: in_world_map
+    detail: bool in_world_map
+    opcode: 0x8170
+    doc: Returns 1 if the player is looking at the world map, or 0 at any other time. Obviously this is only useful in global scripts, since normal scripts will never get the chance to run on the world map.
+  - name: force_encounter
+    detail: void force_encounter(int map)
+    opcode: 0x8171
+    doc: Can be called either from a global script while traveling on the world map, or from a normal script while on a local map. In either case the encounter occurs shortly after the next time the player moves on the world map. The player will not get an outdoorsman skill check.
+  - name: force_encounter_with_flags
+    detail: void force_encounter_with_flags(int map, int flags)
+    opcode: 0x8229
+    doc: Does the same thing as `force_encounter`, but allows the specification of some extra options. Forcing a random encounter on a map that is not normally used for random encounters may cause the player to lose the car, if they have it. In this case use `force_encounter_with_flags` with the `ENCOUNTER_FLAG_NO_CAR` flag set.
+  - name: set_map_time_multi
+    detail: void set_map_time_multi(float multi)
+    opcode: 0x822a
+    doc: Adjusts how fast time passes while you're on the world map. It takes a single float as an argument, where 1 is the normal speed. This function works in addition to the `WorldMapTimeMod` setting in `ddraw.ini` and the Pathfinder perk, rather than overriding it, so calling `set_map_time_multi(0.5)` when the player has 2 levels of Pathfinder would result in time passing at 25% the normal speed on the world map.
+  - name: set_world_map_pos
+    detail: void set_world_map_pos(int x, int y)
+    opcode: 0x8172
+  - name: set_map_enter_position
+    detail: void set_map_enter_position(int tile, int elevation, int rotation)
+    doc: "Overrides the players entry position when entering the map through exit grids. Setting the tile to 0 will put the player on the start hex (default tile and elevation) of the map. Works only in `map_enter_p_proc` procedure."
+  - name: get_map_enter_position
+    detail: array get_map_enter_position()
+    doc: "Returns an array of the player's position data (index: 0 - tile, 1 - elevation, 2 - rotation) when entering the map through exit grids. If entering from the world map, the tile value will be -1. Should be called in `map_enter_p_proc` procedure to get the correct position data."
+
+- name: Get position on map
+  parent: Maps and encounters
+  doc: The mapper manual lists the functions `world_map_x_pos` and `world_map_y_pos`, which supposedly return the player's x and y positions on the world map. `get_world_map_x/y_pos` are included here anyway, because I was unable to get those original functions to work, or even to find any evidence that they existed in game.
```

Comment:
world_map_x_pos/y_pos - таких bis функций не существует.

Reply:
Вот корретные их имена: worldmap_xpos, worldmap_ypos

Reply:
This is what the documentation [states](https://github.com/phobos2077/sfall/blob/250e7bef8a9f83af7af9bf8379524fe03b3ed679/artifacts/scripting/sfall%20function%20notes.txt#L35) in relation to these functions.
I didn't presume to write the documentation for you, only to format it so that it can be machine readable.

Reply:
For some reason I thought some of these comments are new. Nevermind then.

Reply:
I took a few liberties with wording/formatting, but there's no information added from myself, and nothing omitted (unless I missed something).

Reply:
... meaning function documentation, specifically.

I did conjure up a [best practices](https://bgforgenet.github.io/sfall/best-practices/) page entirely from my head. I hope you will improve and expand it.